matrix-org/synapse
# Move scripts directory inside synapse, exposing as setuptools entry_points

Part of #12107.

I haven't done `synctl` here, to try and keep the review on the small side.

I've tested that these scripts are exported in a vaguely reasonable fashion as follows:

- test a wheel
  - pip install build && python -m build
  - cd /tmp && python -m venv env
  - source env/bin/activate
  - pip install /path/to/synapse/dist/*.whl
  - ls env/bin/
  - try running various scripts with --help as a sanity check
- test a docker container
  - DOCKER_BUILDKIT=1 docker build -t matrixdotorg/synapse:latest -f docker/Dockerfile .
  - docker run -it --entrypoint /bin/bash matrixdotorg/synapse:latest
  - ls /usr/local/bin
  - try running various scripts with --help as a sanity check
